controller_manager:
  ros__parameters:
    use_sim_time: True
    update_rate: 100  # Hz

    arm_l_controller:
      type: joint_trajectory_controller/JointTrajectoryController


    arm_r_controller:
      type: joint_trajectory_controller/JointTrajectoryController


    neck_controller:
      type: joint_trajectory_controller/JointTrajectoryController


    body_controller:
      type: joint_trajectory_controller/JointTrajectoryController


    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

arm_l_controller:
  ros__parameters:
    joints:
      - arm_l_joint1
      - arm_l_joint2
      - arm_l_joint3
      - arm_l_joint4
      - arm_l_joint5
      - arm_l_joint6
      - arm_l_joint7
      - l_rh_r1_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 200.0
    action_monitor_rate: 20.0

    allow_partial_joints_goal: false
    open_loop_control: true
    allow_integration_in_goal_trajectories: true
    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 0.0
arm_r_controller:
  ros__parameters:
    joints:
      - arm_r_joint1
      - arm_r_joint2
      - arm_r_joint3
      - arm_r_joint4
      - arm_r_joint5
      - arm_r_joint6
      - arm_r_joint7
      - r_rh_r1_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 200.0
    action_monitor_rate: 20.0

    allow_partial_joints_goal: false
    open_loop_control: true
    allow_integration_in_goal_trajectories: true
    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 0.0
neck_controller:
  ros__parameters:
    joints:
      - neck_joint1
      - neck_joint2
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 200.0
    action_monitor_rate: 20.0

    allow_partial_joints_goal: false
    open_loop_control: true
    allow_integration_in_goal_trajectories: true
    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 0.0
body_controller:
  ros__parameters:
    joints:
      - linear_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 200.0
    action_monitor_rate: 20.0

    allow_partial_joints_goal: false
    open_loop_control: true
    allow_integration_in_goal_trajectories: true
    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 0.0
